packages/block-library/src/template-part/edit/index.js

```diff
@@ -180,3 +220,33 @@ export default function TemplatePartEdit( {
 		</RecursionProvider>
 	);
 }
+
+function PatternsSetup( { blockNameWithArea, clientId, setStartingPattern } ) {
+	const onSelect = ( blocks ) => {
+		const clonedBlocks = blocks.map( ( block ) => cloneBlock( block ) );
+		setStartingPattern( clonedBlocks );
+	};
+
+	return (
+		<BlockPatternSetup
+			blockName={ 'core/template-part' }
+			clientId={ clientId }
+			startBlankComponent={
+				<StartBlankComponent
+					setStartingPattern={ setStartingPattern }
+				/>
+			}
+			onBlockPatternSelect={ onSelect }
+			filterPatternsFn={ ( pattern ) =>
+				pattern?.blockTypes?.some?.(
+					( blockType ) => blockType === blockNameWithArea
+				)
+			}
+		/>
+	);
+}
+
+function StartBlankComponent( { setStartingPattern } ) {
+	setStartingPattern( [] );
```

Comment:
This needs to run only once.

```suggestion
	useEffect( () => {
		setStartingPattern( [] );
	}, [] );
```